### PR TITLE
321: Add new note and examples demonstrating adaptive serialization method

### PR DESCRIPTION
**Pull request by cedporter**
Per [Issue 321](https://github.com/qt4cg/qtspecs/issues/321), I've added a new note and two additional simple examples noting the adaptive serialization method to draw attention to this feature in the existing specs.

### CODE REVIEW

File: specifications/xpath-functions-40/src/function-catalog.xml [hunk 1]
```diff
@@ -16424,6 +16430,10 @@ else
                <fos:result><![CDATA['<a b="3"/>']]></fos:result>
             </fos:test>
          </fos:example>
+         <fos:example>
+            <p>The expression<code>fn:serialize(map{"a":"AB", "b": "BC"}, map{"method":"adaptive"})</code> returns <code>"map{"a":"AB","b":"BC"}"</code></p>
```
Comment (ChristianGruen) [inline specifications/xpath-functions-40/src/function-catalog.xml:16434]:
```suggestion
            <p>The expression <code>fn:serialize(map{"a":"AB", "b": "BC"}, map{"method":"adaptive"})</code> returns <code>"map{"a":"AB","b":"BC"}"</code></p>
```

File: specifications/xpath-functions-40/src/function-catalog.xml [hunk 2]
```diff
@@ -16424,6 +16430,10 @@ else
...
+            <p>The expression<code>fn:serialize(array{"a",3, attribute test {"true"}}, map{"method":"adaptive"})</code> returns <code>"["a",3,test="true"]"</code></p>
```
Comment (ChristianGruen) [inline specifications/xpath-functions-40/src/function-catalog.xml:16435]:
```suggestion
            <p>The expression <code>fn:serialize(array{"a",3, attribute test {"true"}}, map{"method":"adaptive"})</code> returns <code>"["a",3,test="true"]"</code></p>
```

Comment (cedporter) [reply specifications/xpath-functions-40/src/function-catalog.xml:16435]: I think this is fixed in my latest commit. Github's being a little obtuse with it's diffing.

File: specifications/xpath-functions-40/src/function-catalog.xml (hunk 1 above)
Comment (cedporter) [reply specifications/xpath-functions-40/src/function-catalog.xml:16434]: I believe this is addressed in my commit.

**Pull request merged**